project/bmd/models.py

```diff
@@ -1,697 +1,362 @@
+            )
+            self._session = session
 
+        if withModels and not session.has_models:
+            for model in self.models.all():
+                session.add_model(
+                    model.name,
+                    overrides=model.overrides,
+                    id=model.id
+                )
 
-class BMD_Assessment_Settings(models.Model):
-    assessment = models.OneToOneField('assessment.Assessment',
-                                      related_name='BMD_Settings')
-    BMDS_version = models.CharField(max_length=10, choices=BMDS_CHOICES,
-                                    default=max(BMDS_CHOICES)[0])
-    created = models.DateTimeField(auto_now_add=True)
-    last_updated = models.DateTimeField(auto_now=True)
+        return session
 
-    class Meta:
-        verbose_name_plural = "BMD assessment settings"
+    def get_model_options(self):
+        return self.get_session().get_model_options()
 
-    def __unicode__(self):
-        return self.assessment.__unicode__() + ' BMD settings'
+    def get_bmr_options(self):
+        return self.get_session().get_bmr_options()
 
-    def get_absolute_url(self):
-        return reverse('bmd:assess_settings_detail', args=[self.assessment.pk])
+    def get_selected_model(self):
+        return SelectedModel.objects\
+            .filter(endpoint=self.endpoint_id)\
+            .first()
 
-    def get_assessment(self):
-        return self.assessment
+    def get_logic(self):
+        return LogicField.objects\
+            .filter(assessment=self.endpoint.assessment_id)
 
 
-class LogicField(models.Model):
-    """
-    BMD decision-logic selections. Specific to an entire BMD assessment
-    """
-    assessment = models.ForeignKey('assessment.Assessment', related_name='BMD_Logic_Fields', editable=False)
-    created = models.DateTimeField(auto_now_add=True)
-    last_updated = models.DateTimeField(auto_now=True)
-    logic_id = models.PositiveSmallIntegerField(editable=False)
-    name = models.CharField(max_length=30, editable=False)
-    function_name = models.CharField(max_length=25, editable=False)
-    description = models.TextField(editable=False)
-    failure_bin = models.PositiveSmallIntegerField(
-        choices=LOGIC_BIN_CHOICES, blank=False,
-        help_text="If the test fails, select the model-bin should the model be placed into.")
-    threshold = models.DecimalField(
-        max_digits=5, decimal_places=2, null=True, blank=True,
-        help_text="If a threshold is required for the test, threshold can be specified to non-default.")
-    continuous_on = models.BooleanField(default=True, verbose_name="Continuous Datasets")
-    dichotomous_on = models.BooleanField(default=True, verbose_name="Dichotomous Datasets")
-    cancer_dichotomous_on = models.BooleanField(default=True, verbose_name="Cancer Dichotomous Datasets")
+class Model(models.Model):
 
-    class Meta:
-        ordering = ['logic_id']
+    IMAGE_UPLOAD_TO = 'bmds_plot'
```

Comment:
Well, it goes into the `MEDIA` path, which is standard django behavior. I'm surprised that this folder isn't created upon startup; maybe we could just add a check in `bmd/apps.py` in the startup to create this path if it doesn't exist?